### PR TITLE
Add query port status

### PR DESCRIPTION
**Pull request by abulrim**
## Unsure about naming

### CODE REVIEW

File: lib/conduit/telispire/request_mocker/query_port_status.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+require_relative "base"
+
+module Conduit::Telispire::RequestMocker
+  # Request mocker for query_port_status.
+  class QueryPortStatus < Base
+    private
+
+    # add `invalid` requests to the fixtures whitelist
+    def fixtures_whitelist
+      super << :invalid
```
Comment (coreypurcell) [inline lib/conduit/telispire/request_mocker/query_port_status.rb]: why add an invalid? We have the concept of failure that's supposed to represent a failed query port. Error represents the case of the server not responding correctly like a 500 error. I think the use of failure like it's a 500 is incorrect.

Looking at conduit-sprint and conduit-atom show that failure is more like an http: 422. Error is like an http: 500. Unfortunate that we're just now noticing it and not on the past pull requests.

Comment (abulrim) [inline lib/conduit/telispire/request_mocker/query_port_status.rb]: I added an invalid mock to test the 2 kinds of failures: one when the port fails (we get a valid response body that is not successful) and one for failures that are not server errors (faults without a body). Isn't this the desired behavior you suggested in your comment?

Comment (coreypurcell) [inline lib/conduit/telispire/request_mocker/query_port_status.rb]: I think I understand now.

**Comment by abulrim** [comment]
Updated: renamed to `submit_port_validation`

**Pull request merged**